src/maestro/file_logger.py

```diff
@@ -0,0 +1,37 @@
+import uuid
+from datetime import datetime, UTC
+from pathlib import Path
+
+class FileLogger:
+    def __init__(self, log_dir=None):
+        self.log_dir = Path(log_dir) if log_dir else Path.cwd()
+        self.log_dir.mkdir(parents=True, exist_ok=True)
+
+    def generate_workflow_id(self):
+        return uuid.uuid4().hex
+
+    def log_workflow_run(self, workflow_id, workflow_name, prompt, output, models_used, status):
```

Comment:
question about the log formatting: will you be able to read these files into something like pandas or a database without too much work if you want to run analysis on them?